indico/web/client/js/custom_elements/ind_date_picker.js

```diff
@@ -5,47 +5,71 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
-import preval from 'preval.macro';
-
 import CustomElementBase from 'indico/custom_elements/_base';
+import {
+  DateRange,
+  OpenDateRange,
+  SparseDateRange,
+  getToday,
+  isSameDate,
+  toDateString,
+} from 'indico/utils/date';
 import {formatDate} from 'indico/utils/date_format';
 import {createDateParser} from 'indico/utils/date_parser';
+import {getWeekInfoForLocale, getFirstDayOfWeek, getWeekdayNames} from 'indico/utils/l10n';
 import * as positioning from 'indico/utils/positioning';
+import {setNativeInputValue} from 'indico/utils/react_integration';
+
+import * as ds from './date_selection';
 
 import './ind_date_picker.scss';
 
 const DEFAULT_LOCALE = document.documentElement.dataset.canonicalLocale;
-const CALENDAR_CELL_COUNT = 42; // 6 rows, 7 days each
+const KEYBOARD_MOVEMENT = {
+  ArrowRight: 'nextday',
+  ArrowLeft: 'previousday',
+  ArrowDown: 'nextweek',
+  ArrowUp: 'previousweek',
+};
+// States used when determining the next action in range picker
+const START_NEEDED = 0;
+const END_NEEDED = 1;
+const BOTH_POPULATED = 2;
 
 customElements.define(
   'ind-date-picker',
   class extends CustomElementBase {
     // Last value of the internal id used in the widgets
     static lastId = 0;
 
+    static attributes = {
+      min: Date, // Minimum selectable date (inclusive)
+      max: Date, // Maximum selectable date (inclusive)
+      format: String, // Date format
+    };
+
+    static observedAttributes = ['min', 'max'];
+
     get value() {
       return this.querySelector('input').value;
     }
 
     setup() {
+      const indDatePicker = this;
       const id = `date-picker-${this.constructor.lastId++}`;
       const input = this.querySelector('input');
       const openCalendarButton = this.querySelector('button');
       const formatDescription = this.querySelector('[data-format]');
       const indCalendar = this.querySelector('ind-calendar');
 
-      const dateFormat = formatDescription.textContent
-        .split(':')[1]
-        .trim()
-        .toLowerCase();
-      const parseDate = createDateParser(dateFormat);
+      const parseDate = createDateParser(this.format);
 
-      indCalendar.format = dateFormat;
-      indCalendar.value = toDateString(parseDate(this.value));
+      indCalendar.format = this.format;
+      indCalendar.rangeStart = indCalendar.rangeEnd = toDateString(parseDate(this.value));
+      indCalendar.min = this.min;
+      indCalendar.max = this.max;
```

Comment:
I think something around here broke: When using the single DatePicker, it no longer applies a minimum value (did not test w/ max, but probably similarly broken).

I've pushed a test commit to your branch that uses the new pickers in Room Booking, so you can easily test there. For the ranges everything works as expected, but for the single picker not so much.

Reply:
This should be fixed now.